File: docs/examples/struct-example-2.js
import { User } from './struct-example-2.zig';

const user = new User({
    id: 1234n,
    name: "Bigus Dickus",
    email: "madeupname12@rome.gov.it",
    age: 32,
    address: null,
});
user.address = {
    street: '1 Colosseum Sq.',
    city: 'Rome',
    state: 'NY',
    zipCode: '10001',
};
user.print();

// calling method like regular function
User.print({
    id: 1234n,
    name: "Bigus Dickus",
    email: "madeupname12@rome.gov.it",
    age: 32,
    address: {
        street: '1 Colosseum Sq.',
        city: 'Rome',
        state: 'NY',
        zipCode: '10001',
    },
});

File: docs/examples/struct-example-2.zig
const std = @import("std");

pub const Address = struct {
    street: []const u8,
    city: []const u8,
    state: [2]u8,
    zipCode: u32,
};
pub const User = struct {
    id: u64,
    name: []const u8,
    email: []const u8,
    age: ?u32 = null,
    popularity: i64 = -1,
    address: ?Address,

    pub fn print(self: User) void {
        std.debug.print("Name: {s}\n", .{self.name});
        std.debug.print("E-mail: {s}\n", .{self.email});
        if (self.age) |age| {
            std.debug.print("Age: {d}\n", .{age});
        }
        std.debug.print("Popularity: {d}\n", .{self.popularity});
        if (self.address) |address| {
            std.debug.print("Street: {s}\n", .{address.street});
            std.debug.print("City: {s}\n", .{address.city});
            std.debug.print("State: {s}\n", .{address.state});
            std.debug.print("ZIP code: {d}\n", .{address.zipCode});
        }
    }
};
